src/load_inputs/load_generators_data.jl

```diff
@@ -185,44 +185,45 @@ function load_generators_data!(setup::Dict, path::AbstractString, inputs_gen::Di
 		inputs_gen["C_Start"] = zeros(Float64, G, inputs_gen["T"])
-		gen_in[!,:CO2_per_Start] = zeros(Float64, G)
 	end
 
-	# Heat rate of all resources (million BTUs/MWh)
-	heat_rate = convert(Array{Float64}, gen_in[!,:Heat_Rate_MMBTU_per_MWh])
-	# Fuel used by each resource
-	fuel_type = gen_in[!,:Fuel]
-	# Maximum fuel cost in $ per MWh and CO2 emissions in tons per MWh
-	inputs_gen["C_Fuel_per_MWh"] = zeros(Float64, G, inputs_gen["T"])
-	gen_in[!,:CO2_per_MWh] = zeros(Float64, G)
+	# scale the start costs 
 	for g in 1:G
-		# NOTE: When Setup[ParameterScale] =1, fuel costs are scaled in fuels_data.csv, so no if condition needed to scale C_Fuel_per_MWh
-		inputs_gen["C_Fuel_per_MWh"][g,:] = fuel_costs[fuel_type[g]].*heat_rate[g]
-		gen_in[g,:CO2_per_MWh] = fuel_CO2[fuel_type[g]]*heat_rate[g]
-		gen_in[g,:CO2_per_MWh] *= scale_factor
-		# kton/MMBTU * MMBTU/MWh = kton/MWh, to get kton/GWh, we need to mutiply 1000
 		if g in inputs_gen["COMMIT"]
-			# Start-up cost is sum of fixed cost per start plus cost of fuel consumed on startup.
-			# CO2 from fuel consumption during startup also calculated
-
-			inputs_gen["C_Start"][g,:] = gen_in[g,:Cap_Size] * (fuel_costs[fuel_type[g]] .* start_fuel[g] .+ start_cost[g])
-			# No need to re-scale C_Start since Cap_size, fuel_costs and start_cost are scaled When Setup[ParameterScale] =1 - Dharik
-			gen_in[g,:CO2_per_Start]  = gen_in[g,:Cap_Size]*(fuel_CO2[fuel_type[g]]*start_fuel[g])
-			gen_in[g,:CO2_per_Start] *= scale_factor
-			# Setup[ParameterScale] =1, gen_in[g,:Cap_Size] is GW, fuel_CO2[fuel_type[g]] is ktons/MMBTU, start_fuel is MMBTU/MW,
-			#   thus the overall is MTons/GW, and thus gen_in[g,:CO2_per_Start] is Mton, to get kton, change we need to multiply 1000
-			# Setup[ParameterScale] =0, gen_in[g,:Cap_Size] is MW, fuel_CO2[fuel_type[g]] is tons/MMBTU, start_fuel is MMBTU/MW,
-			#   thus the overall is MTons/GW, and thus gen_in[g,:CO2_per_Start] is ton
+			# Start-up cost is sum of fixed cost per start startup.
+			inputs_gen["C_Start"][g,:] .= gen_in[g,:Cap_Size] * ( start_cost[g])
 		end
 	end
 
 	load_vre_stor_data!(inputs_gen, setup, path)
+
+	
+    # write zeros if col names are not in the gen_in dataframe
+	missing_cols_for_co2 = ["BIOMASS", "CO2_Capture_Rate", "CO2_Capture_Rate_Startup", "CCS_Disposal_Cost_per_Metric_Ton"]
+	for col in missing_cols_for_co2
+		ensure_column!(gen_in, col, 0)
+	end
+
+    # Scale CCS_Disposal_Cost_per_Metric_Ton for CCS units 
+	gen_in.CCS_Disposal_Cost_per_Metric_Ton /= scale_factor
+
+	# get R_ID when fuel is not None
+	inputs_gen["FUEL"] = gen_in[(gen_in[!,:Fuel] .!= "None"),:R_ID]
+
+    # Piecewise fuel usage module
+	# Users should specify how many segements are used to build piecewise fuel comsumption.
+	# Users should at least provide Slope1 and Intercept1 if they want to use piecewise fuel usage
+	if setup["UCommit"] > 0
+        if setup["PiecewiseFuelUsage"] == 1
+			process_piecewisefuelusage!(inputs_gen, gen_in, scale_factor)
+		end
+	end
+
+
+
```

Comment:
Please normalize tabs to 4 spaces here.
Indentations should be 4 spaces.
Comments should have the same indentation level as the code.